### ISSUE TITLE
[Bug]: Switch to frame by element

### ISSUE THREAD

**Issue by ephung01** [microsoft/playwright-java#560]
### Playwright version

1.13.0 (existing in older version too)

### Operating system

Windows

### What browsers are you seeing the problem on?

Chromium

### Other information

Java 1.8
windows 10

### What happened? / Describe the bug

When use switch to iframe by element.  It seem switch to a non-existing iframe (see pix)

The page have 6 iframes but in debug windows show there are 7 iframes

Page source:
![image](https://user-images.githubusercontent.com/6579909/129377632-7909762a-184b-44ee-a2b6-04ba4a45a00a.png)
Debug Windows:
![image](https://user-images.githubusercontent.com/6579909/129377432-8da75bc9-32b7-4f36-a64a-c7b142243bcd.png)

Frame nestedFrame = elementHandle.contentFrame();
![image](https://user-images.githubusercontent.com/6579909/129380706-2e2c4225-9a6e-4c7b-91c2-0ee9dc1f7d98.png)

**Comment by ephung01** [comment]
Why Frame object has no ID attribute?

**Comment by yury-s** [comment]
So what is the issue? Doe playwright return a frame which is not in the tree?

**Comment by ephung01** [comment]
Hello Yury.
Sorry I've out for awhile due to too many production releases.
I find out the issue on Iframe.
The logic : the iframe were in part of the initial rendering html and the iframe's id and name are updated through javascript later on.  When javascript updated, it seems like Playwright creates a new iframe instead of update an existing one.   So when I switch to the iframe (by Element) it seems switch to the old one and got stuck there.

In short, the Playwright's Element is updated correctly, but seem likes Frames Object is not.

**Comment by ephung01** [comment]
> 
> 
> So what is the issue? Doe playwright return a frame which is not in the tree?

The frame is in the tree, but it seem like empty iframe

**Comment by ephung01** [comment]
Let me get more detail.   Sorry, I might have missed diagnosed it.
